doc/choose-a-license.md

```diff
@@ -0,0 +1,312 @@
+
+### Copyleft versus Permissive
+
+"Copyleft" and "Permissive" are terms used to compare legal philosophies and attributes of FOSS licenses to traditional proprietary licenses.
+
+- **Copyleft:** A license that **not only grants the freedoms** mentioned above but furthermore **requires that those freedoms apply to any derivative works**. Requiring as a condition of the license that any derivative works also be distributed under the same licensing terms as the original, these copyleft licensing terms are also sometimes referred to as reciprocity or "share-alike" provisions, and because of these reciprocity requirements, copyleft licenses are considered "restrictive" licenses, though these restrictions guarantee perpetual open access.
+
+**Examples** of popular copyleft FOSS licenses include the **GNU General Public License (GPL)**, **GNU Lesser General Public License (LGPL)**, and the **Mozilla Public License (MPL)**. The GNU Licenses are the most well known of all the FOSS licenses and have a strong community of supporters and advocates. Of these, the GPL has the strongest reciprocity requirements and is considered a "strong" copyleft license. The LGPL (the "Lesser GPL," denoting its weaker copyleft requirements) is very similar to the GPL from which it is derived, but allows for linking to proprietary code under certain circumstances. Similarly, the MPL allows copyleft to be applied to some parts of the code and not others. The LGPL and MPL are considered a compromise between the strong copyleft of GPL and permissive licenses such as the BSD/MIT.
+
+- **Permissive or non-copyleft:** A license that **grants the freedoms** mentioned above but that **does not have a clause requiring that they apply to derivative works as well**. Permissively licensed software may be incorporated into "closed" proprietary programs with no requirement that the source code be disclosed if the combined software is distributed.
+
+**Examples** of popular permissive FOSS licenses include the **Berkeley Software Distribution (BSD)**, **MIT** and **Apache** licenses. The BSD and MIT licenses are often mentioned interchangeably due to very similar language and terms that accomplish largely identical goals. The primary intent of these licenses is to allow the use, distribution, and modification of your code for any purpose, while making sure that you as the creator receive credit for your work. The Apache license is similar in effect to the BSD/MIT, but include a license for patents related to the software (this can be desirable or not, depending on the situation).
+
+Below is a list of licenses in descending order **from strong copyleft at the top to completely non-copyleft at the bottom**:
+
+- **GNU General Public License version 3** [GPL-3.0](https://www.gnu.org/licenses/gpl.html)
+- **GNU Affero General Public License version 3** [AGPL-3.0](https://www.gnu.org/licenses/agpl.html)
```

Comment:
AGPL is stronger than GPL, should be sorted first.

Reply:
Thanks for the feedback @da2ce7

I've just fixed it.